src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -406,7 +406,7 @@ internal sealed partial class BoundCompoundAssignmentOperator
     internal sealed partial class BoundParameter
     {
         public BoundParameter(SyntaxNode syntax, ParameterSymbol parameterSymbol, bool hasErrors = false)
-            : this(syntax, parameterSymbol, parameterSymbol.Type, hasErrors)
+            : this(syntax, parameterSymbol, parameterSymbol.Type, hasErrors || parameterSymbol.Type.IsErrorType())
```

Comment:
>|| parameterSymbol.Type.IsErrorType() [](start = 76, length = 37)

I do not understand the motivation for the changes in the second commit and I am not sure they are necessary.

Reply:
Discussed offline. There are ~200 tests failing because of the fix, and most of them just need updating the baseline. I'm reviewing the rest now.